internal/service/alarms/internal/alarms_server.go

```diff
@@ -191,8 +195,58 @@ func (a *AlarmsServer) GetAlarm(ctx context.Context, request api.GetAlarmRequest
 }
 
 func (a *AlarmsServer) AckAlarm(ctx context.Context, request api.AckAlarmRequestObject) (api.AckAlarmResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	// Fetch the Alarm Event Record to be acknowledged
+	record, err := a.AlarmsRepository.GetAlarmEventRecord(ctx, request.AlarmEventRecordId)
+	if errors.Is(err, utils.ErrNotFound) {
+		// Nothing found
+		return api.AckAlarm404ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			AdditionalAttributes: &map[string]string{
+				"alarmEventRecordId": request.AlarmEventRecordId.String(),
+			},
+			Detail: "requested Alarm Event Record not found",
+			Status: http.StatusNotFound,
+		}), nil
+	}
+	if err != nil {
+		return nil, fmt.Errorf("failed to get Alarm Event Record: %w", err)
+	}
+
+	alarmAcknowledged := request.Body.AlarmAcknowledged
+
+	// Check if requested Acknowledged status is true
+	if !alarmAcknowledged {
+		// Bad request - Acknowledged status is expected to be true
+		return api.AckAlarm400ApplicationProblemPlusJSONResponse(common.ProblemDetails{
+			AdditionalAttributes: &map[string]string{
+				"alarmEventRecordId": request.AlarmEventRecordId.String(),
+			},
+			Detail: "Alarm Event Record Acknowledged status is expected to be true",
+			Status: http.StatusBadRequest,
+		}), nil
+	}
```

Comment:
We may want to consider allowing un-acknowledging an alarm so that a client can 'undo' a mistake of acknowledging the wrong alarm.  The interface document doesn't describe how to handle this so a more flexible approach may be better.

Reply:
That was my initial thought as well, but then I realized that the specifications refer to this operation in only one direction. Either acknowledging or clearing the alarm but no the reverse operation. The use case in O-RAN.WG6.ORCH-USE-CASES-R003-v11.00 too. In fact, for clearing the alarm, perceivedSeverity is restricted to 5, so once you set it, there is no way back. I think it is ok to not allow the reverse operation for acknowledging. Functionality is not affected, it is only a way to mark that an alarm has been seen. Also, alarms will be eventually deleted. So, this is not a persistent object.

Reply:
ok